numpy/random/mtrand/mtrand.pyx

```diff
@@ -1094,6 +1094,8 @@ cdef class RandomState:
 
         # Format and Verify input
         a = np.array(a, copy=False)
+        if np.prod(size) == 0:
```

Comment:
This isn't testing the right thing at all. The code already worked when `size` had zero-entries, and this breaks `np.random.choice(['a', 'b'], size=(3, 0, 4))`